numba/targets/linalg.py

```diff
@@ -667,3 +688,300 @@ def inv(context, builder, sig, args):
+                               ldvr)
+
+            # put these in to help with liveness analysis,
+            # `.ctypes` doesn't keep the vars alive
+            acpy.size
+            vl.size
+            vr.size
+            w.size
+            return (w, vr.T)
+
+        if isinstance(a.dtype, types.scalars.Complex):
+            return cmplx_eig_impl
+        else:
+            return real_eig_impl
+
+    @overload(numpy.linalg.svd)
+    def svd_impl(a, full_matrices=1):
+        ensure_lapack()
+
+        _check_linalg_matrix(a, "svd")
+
+        F_layout = a.layout == 'F'
+
+        # convert typing floats to numpy floats for use in the impl
+        s_type = getattr(a.dtype, "underlying_float", a.dtype)
+        if s_type.bitwidth == 32:
+            s_dtype = numpy.float32
+        else:
+            s_dtype = numpy.float64
+
+        numba_ez_gesdd_sig = types.intc(
+            types.char,  # kind
+            types.char,  # jobz
+            types.intp,  # m
+            types.intp,  # n
+            types.CPointer(a.dtype),  # a
+            types.intp,  # lda
+            types.CPointer(s_type),  # s
+            types.CPointer(a.dtype),  # u
+            types.intp,  # ldu
+            types.CPointer(a.dtype),  # vt
+            types.intp  # ldvt
+        )
+
+        numba_ez_gesdd = types.ExternalFunction("numba_ez_gesdd",
+                                                numba_ez_gesdd_sig)
+
+        kind = ord(get_blas_kind(a.dtype, "svd"))
+
+        JOBZ_A = ord('A')
+        JOBZ_S = ord('S')
+
+        def svd_impl(a, full_matrices=1):
+            n = a.shape[-1]
+            m = a.shape[-2]
+
+            if not numpy.isfinite(a).all():
+                raise numpy.linalg.LinAlgError(
+                    "Array must not contain infs or NaNs.")
+
+            if F_layout:
+                acpy = numpy.copy(a)
+            else:
+                acpy = numpy.asfortranarray(a)
+
+            ldu = m
+            minmn = min(m, n)
+
+            if full_matrices:
+                JOBZ = JOBZ_A
+                ucol = m
+                ldvt = n
+            else:
+                JOBZ = JOBZ_S
+                ucol = minmn
+                ldvt = minmn
+
+            u = numpy.zeros((ucol, ldu), dtype=a.dtype)
+            s = numpy.zeros(minmn, dtype=s_dtype)
+            vt = numpy.zeros((n, ldvt), dtype=a.dtype)
+
+            r = numba_ez_gesdd(
+                kind,  # kind
+                JOBZ,  # jobz
+                m,  # m
+                n,  # n
+                acpy.ctypes,  # a
+                m,  # lda
+                s.ctypes,  # s
+                u.ctypes,  # u
+                ldu,  # ldu
+                vt.ctypes,  # vt
+                ldvt          # ldvt
+            )
+
+            # help liveness analysis
+            acpy.size
+            vt.size
+            u.size
+            s.size
+
+            # TODO: this logic, no point in churning out a copy if no need
```

Comment:
The implementation is correct, it was a performance thought. Coming back to it, it is probably not be worth it as it is not that likely to happen.